### PR TITLE
Main feature

### LINKED ISSUE #2: [5 pts] Back-End: Create reservations controller
1. Implement add create a new reservation, and list user reservations.
 2.  Check the reservation date to create new reservation

### LINKED ISSUE #3: [5 pts] Back-End: Create API endpoint.
Create all API  end-points , lie add routes

### LINKED ISSUE #9: [5 pts] Back-End: Create Jet Controller
Create  list, create, delete and update  jets controller action methods

### LINKED ISSUE #12: [3.5 pts] Back-End: Create User, Jet and Reservation migrations and models
Create a new User table with the proper migration and model:
   - Name:string

Create a new Jets table with the proper migration and model:
   - Name:string
   - Price_per_day:integer
   - Description:text
   - Size:text
   - Category:text
   - Finance_fee:integer
   - Image:link

Create a new Reservation table with the proper migration and model:
   - User_id
   - Jet_id
   - starting_day:date
   - finish_day:date
   - City:

### LINKED ISSUE #14: [3.5 pts] Back-End: Create associations between tables
Add association between tables on the models.
User has_many reservations.
Jet has_many Reservations and has_many users through reservations.
Reservation belongs_to User and Jet.

### PR DESCRIPTION
**Pull request by masangana**
## Todo List:

✅Back-End: Create User, Jet and Reservation migrations and models
✅Back-End: Create associations between tables
✅Back-End: Create API endpoint.
✅Back-End: Create reservations controller
✅Back-End: Create Jet Controller

closes #12 
closes #3 
closes #14 
closes #2 
closes #9 

Screenshot (331)

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class User < ApplicationRecord
+end
```
Comment (mntedev) [inline app/models/user.rb]: As the reservation belongs to a user, the user should have many reservations. you can add "has_many reservations" in your user model. Many users may reserve each jet at different times. you already add "has_many users through reservations" in your jet model. It is recommended to add additional has many relationships in your user model like this " has_many :jets, through: :reservations"

**Review by mntedev — CHANGES_REQUESTED** [review]
**Hello @masangana** :hand: 
 You did a good job so far :clap: ,   There are some issues that you still need to work on to go to the next project. You are almost there! :muscle: 
image

**Highlights :rocket:** 
  :heavy_check_mark:  No linters error
 :heavy_check_mark:  Professional pull request
:heavy_check_mark:  Good features implementation

### 
 **Project Status : Change Required :recycle: :**  

Kindly check the comments under the review. Implementing those comments makes your project work correctly.
Cheers :wine_glass:  and Happy coding! :computer: 

### **Optional suggestions**

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/models/user.rb (hunk 1 above)
Comment (masangana) [reply app/models/user.rb]: Fixed

**Review by mntedev — APPROVED** [review]
**Hello @masangana :hand:** 
You have done a good job so far, your project is complete :100:  there is nothing to say other than it is time to merge! :ok: 
image

### **Highlights :rocket:**
  Your project is  free from linters error :heavy_check_mark: 
  Implemented the required change correctly :heavy_check_mark: 
 Descriptive  pull request :heavy_check_mark: 

 ### **Project status: Approved :heavy_check_mark:** 
Cheers :wine_glass: and Happy coding! :computer: 

### **Optional suggestions**

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.



As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**